### ISSUE TITLE
hlint ignores pragma to ignore within a module

### ISSUE THREAD

**Issue by EdmundsEcho** [ndmitchell/hlint#988]
I truly enjoy working with `hlint`. 

I have not been successful at suppressing messages. I have also found the documentation in this regard to be a bit confusing when trying to troubleshoot.  For instance , the different between `ANN` and others?

### Issue
The following entry:
```haskell
{-# ANN module ("HLint: ignore Orphan instance"::String) #-}
```
Does not suppress the `[W] Orphan instance` message.  

### Context sensitive
When I run `hlint` from the terminal, it does not show me this warning even when the pragma is removed from the file.

When I run `hlint` using an `HIE` and `coc.nvim` setting, I always get the warning.

The `.hlint.yaml` file only exists in my user directory.  It is the one that is generated by default and does not specify anything much, but certainly nothing in regards to this message.

Thank you in advance for helping me to troubleshoot this issue.

\- E

**Comment by ndmitchell** [comment]
Orphan instances aren't a message that HLint gives, but one that comes from GHC. See https://functor.tokyo/blog/2016-09-16-options-ghc-fno-warn-orphans for how to suppress them. Does that explain your confusion as to why you don't get them from the `hlint` binary etc?
